File: revelare/revelare/report/production_per_day_report/production_per_day_report.py
# Copyright (c) 2019, SHS and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
    columns = get_columns(filters)

    data = obtener_items(filters)

    # chart = get_chart_data(columns)

    # return columns, data, None, chart
    return columns, data


def get_columns(filters):
    '''Genera las columnas necesarias para el reporte'''

    columns = [
        {
            "fieldname": "item_code",
            "label": _("Item Code"),
            "fieldtype": "Link",
            "options": "Item",
            "width": 200
        },
        {
            "fieldname": "item_name",
            "label": _("Item Name"),
            "fieldtype": "Data",
            "width": 200
        },
        {
            "fieldname": "stock_uom",
            "label": _("UOM"),
            "fieldtype": "Data",
            "width": 200
        },
        {
            "fieldname": "qty",
            "label": _("Quantity"),
            "fieldtype": "Data",
            "width": 200
        },
        {
            "fieldname": "actual_qty",
            "label": _("Actual Quantity"),
            "fieldtype": "Data",
            "width": 200
        }
    ]

    return columns


def get_chart_data(columns):
    '''retorna la data necesaria para mostrar las graficas en el reporte'''

    labels = [d.get("label") for d in columns]
    # labels = ["Item Code"]
    chart = {
        "data": {
            'labels': labels,
            'datasets':[]
        }
    }
    chart["type"] = "line"

    return chart


def obtener_items(filters):
    '''Obtiene items de la base de datos en base a los filtros
    '''
    data = []

    # Obtiene la data necesaria
    master_items = []

    production_plans = frappe.db.get_values('Production Plan',
                                            filters={'company': filters.get("company"),
                                                    'posting_date': filters.get("from_date")},
                                            fieldname=['name'], as_dict=1)

    for item in production_plans:
        required_items = frappe.db.sql('''SELECT item_code, item_name, quantity, actual_qty
                                          FROM `tabMaterial Request Plan Item`
                                          WHERE parent=%(plan_name)s''', 
                                      {'plan_name': str(item.get('name'))}, as_dict=True)

        master_items.extend(required_items)

    # Prepara la data para ser mostrada correctamente en el informe
    for item_data in master_items:
        row = frappe._dict({
            "item_code": item_data.item_code,
            "item_name": item_data.item_name,
            "stock_uom": "Lbs",
            "qty": item_data.quantity,
            "actual_qty": item_data.actual_qty
            # "indent": flt(1)
        })

        data.append(row)

    return data

File: revelare/revelare/report/production_per_day_report/production_per_day_report.js
// Copyright (c) 2019, SHS and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Production Per Day Report"] = {
	"filters": [
		{
			fieldname: "company",
			label: __("Company"),
			fieldtype: "Link",
			options: "Company",
			default: frappe.defaults.get_user_default("Company"),
			reqd: 1
		},
		{
			fieldname: "from_date",
			label: __("Date"),
			fieldtype: "Date",
			default: frappe.defaults.get_global_default("year_start_date"),
			reqd: 1
		}
	],
	after_datatable_render: function (datatable_obj) {
		$(datatable_obj.wrapper).find(".dt-row-0").find('input[type=checkbox]').click();
	},
	get_datatable_options(options) {
		return Object.assign(options, {
			checkboxColumn: true,
			events: {
				onCheckRow: function (data) {
					row_name = data[2].content;
					row_values = data.slice(3).map(function (column) {
						return column.content;
					})
					entry = {
						'name': row_name,
						'values': row_values
					}

					let raw_data = frappe.query_report.chart.data;
					let new_datasets = raw_data.datasets;

					var found = false;

					for (var i = 0; i < new_datasets.length; i++) {
						if (new_datasets[i].name == row_name) {
							found = true;
							new_datasets.splice(i, 1);
							break;
						}
					}

					if (!found) {
						new_datasets.push(entry);
					}

					let new_data = {
						labels: raw_data.labels,
						datasets: new_datasets
					}

					setTimeout(() => {
						frappe.query_report.chart.update(new_data)
					}, 500)


					setTimeout(() => {
						frappe.query_report.chart.draw(true);
					}, 1000)

					frappe.query_report.raw_chart_data = new_data;
				},
			}
		});
	}
}
